### PR TITLE
refactor CDC

### PR DESCRIPTION
**Pull request by leechor**

### CODE REVIEW

File: dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/doris/DorisSchemaEvolutionSinkBuilder.java [hunk 1]
```diff
@@ -94,127 +88,126 @@ public DataStreamSource build(
...
+                        executionBuilder.setDeletable(true);
+                    }
+
+                    if (sink.containsKey(DorisSinkOptions.SINK_LABEL_PREFIX.key())) {
+                        executionBuilder.setLabelPrefix(
+                                String.format("%s-%s_%s", sink.get(DorisSinkOptions.SINK_LABEL_PREFIX.key()),
+                                        getSinkSchemaName(table), getSinkTableName(table)));
+                    } else {
+                        executionBuilder.setLabelPrefix(
+                                String.format("dlink-%s_%s%s", getSinkSchemaName(table), getSinkTableName(table),
```
Comment (leechor) [inline dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/doris/DorisSchemaEvolutionSinkBuilder.java]: should changed?

File: dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/doris/DorisSinkBuilder.java [hunk 2]
```diff
@@ -146,33 +156,27 @@ public void addSink(
...
-                            + getSinkTableName(table));
+                    String.format("%s-%s_%s", sink.get(DorisSinkOptions.SINK_LABEL_PREFIX.key()), getSinkSchemaName(table), getSinkTableName(table)));
         } else {
             executionBuilder.setLabelPrefix(
-                    "dlink-"
-                            + getSinkSchemaName(table)
-                            + "_"
-                            + getSinkTableName(table)
-                            + UUID.randomUUID());
+                    String.format("dlink-%s_%s%s", getSinkSchemaName(table), getSinkTableName(table), UUID.randomUUID()));
```
Comment (leechor) [inline dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/doris/DorisSinkBuilder.java]: here

File: dinky-cdc/dinky-cdc-core/src/main/java/org/dinky/cdc/CDCBuilderFactory.java [hunk 3]
```diff
@@ -27,34 +27,31 @@
...
-                        () -> {
-                            throw new FlinkClientException(
-                                    "未匹配到对应 CDC Source 类型的【" + config.getType() + "】。");
-                        })
-                .get()
-                .create(config);
+
+        Supplier<CDCBuilder> cdcBuilderSupplier = CDC_BUILDER_MAP.get(config.getType());
+        if (cdcBuilderSupplier == null) {
+            throw new FlinkClientException("未匹配到对应 CDC Source 类型的【" + config.getType() + "】。");
```
Comment (Zzm0809) [inline dinky-cdc/dinky-cdc-core/src/main/java/org/dinky/cdc/CDCBuilderFactory.java]: use English ??

File: dinky-cdc/dinky-cdc-core/src/main/java/org/dinky/cdc/SinkBuilderFactory.java [hunk 4]
```diff
@@ -25,27 +25,26 @@
...
-                }
-            };
+    private static final Map<String, Supplier<SinkBuilder>> SINK_BUILDER_MAP =
+            ImmutableMap.of(
+                    SQLSinkBuilder.KEY_WORD, SQLSinkBuilder::new,
+                    SQLCatalogSinkBuilder.KEY_WORD, SQLCatalogSinkBuilder::new);
 
     public static SinkBuilder buildSinkBuilder(FlinkCDCConfig config) {
         if (Asserts.isNull(config) || Asserts.isNullString(config.getSink().get("connector"))) {
             throw new FlinkClientException("请指定 Sink connector。");
```
Comment (Zzm0809) [inline dinky-cdc/dinky-cdc-core/src/main/java/org/dinky/cdc/SinkBuilderFactory.java]: use English ??

File: dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/SinkBuilderFactory.java [hunk 5]
```diff
@@ -29,33 +30,30 @@
...
+                    .put(SQLCatalogSinkBuilder.KEY_WORD, SQLCatalogSinkBuilder::new)
+                    .put(KafkaSinkBuilder.KEY_WORD, KafkaSinkBuilder::new)
+                    .put(DorisSinkBuilder.KEY_WORD, DorisSinkBuilder::new)
+                    .put(DorisExtendSinkBuilder.KEY_WORD, DorisExtendSinkBuilder::new)
+                    .put(DorisSchemaEvolutionSinkBuilder.KEY_WORD, DorisSchemaEvolutionSinkBuilder::new)
+                    .build();
 
     public static SinkBuilder buildSinkBuilder(FlinkCDCConfig config) {
         if (Asserts.isNull(config) || Asserts.isNullString(config.getSink().get("connector"))) {
             throw new FlinkClientException("请指定 Sink connector。");
```
Comment (Zzm0809) [inline dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/SinkBuilderFactory.java]: change to English

Comment (leechor) [reply dinky-cdc/dinky-cdc-plus/src/main/java/org/dinky/cdc/SinkBuilderFactory.java]: resolved

**Comment by melin** [comment]
如果同步表太多，sink task 数量太多，有考虑合并sink 吗？

**Comment by leechor** [comment]
> 如果同步表太多，sink task 数量太多，有考虑合并sink 吗？

这个PR属于0.8重构CDC部分代码组织结构,不涉及大的功能调整, 可以开个issue,与@aiwenmo 讨论

File: dinky-metadata/dinky-metadata-oracle/src/main/java/org/dinky/metadata/driver/OracleDriver.java [hunk 6]
```diff
@@ -186,20 +184,4 @@ public String getCreateTableSql(Table table) {
...
-        ds.setUsername(config.getUsername());
-        ds.setPassword(config.getPassword());
-        ds.setValidationQuery("select 1 from dual");
-        ds.setTestWhileIdle(true);
-        ds.setBreakAfterAcquireFailure(true);
-        ds.setFailFast(true);
-        ds.setInitialSize(1);
-        ds.setMaxActive(8);
-        ds.setMinIdle(5);
-    }
```
Comment (aiwenmo) [inline dinky-metadata/dinky-metadata-oracle/src/main/java/org/dinky/metadata/driver/OracleDriver.java:204]: Why delete it?

Comment (leechor) [reply dinky-metadata/dinky-metadata-oracle/src/main/java/org/dinky/metadata/driver/OracleDriver.java:204]: > Why delete it?

move up, not delete

Comment (leechor) [reply dinky-metadata/dinky-metadata-oracle/src/main/java/org/dinky/metadata/driver/OracleDriver.java:204]: add validationQuery class member.

**Review by aiwenmo — APPROVED** [review]
LGTM

**Pull request merged**